src/infrastructure/middleware/localization/index.ts

```diff
@@ -1,19 +1,35 @@
 import { LocaleTypeEnum } from "../../../application/shared/locals/LocaleType.enum";
 import { IRequest } from "../../../adapters/controllers/base/Base.controller";
 import { Request, Response, NextFunction } from "../../app/core/Modules";
-import { DefaultValue } from "../../../domain/shared/utils/DefaultValue";
 import appMessages from "../../../application/shared/locals/messages";
 import { TypeParser } from "../../../domain/shared/utils/TypeParser";
 import appWords from "../../../application/shared/locals/words";
 import { Middleware } from "../types";
 import config from "../../config";
 
+const getLanguage = (req: Request) => {

```

Comment:
Please include your function as private static in the class context to avoid starting to mix programming styles.
Additionally, try to use `Object.hasOwn()` or `Reflect` to validate the existence of the property.
```ts
private static getLanguage(req: Request): LocaleTypeEnum {
// ...
  if (Object.hasOwn(languages, key)) {
  // Or
  if (Reflect.has(languages, key)) {
```

Reply:
adjusted, hope I understood